test/promise.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const buildType = process.config.target_defaults.default_configuration;
+const assert = require('assert');
+
+test(require(`./build/${buildType}/binding.node`));
+test(require(`./build/${buildType}/binding_noexcept.node`));
+
+function test(binding) {
+  assert.strictEqual(binding.promise.isPromise({}), false);
+
+  let resolved = null;
+  let resolving = binding.promise.resolvePromise('resolved');
+  assert.strictEqual(binding.promise.isPromise(resolving), true);
+
+  resolving.then(function(value) {
+    resolved = value;
```

Comment:
Using [`common.mustNotCall()`](https://github.com/nodejs/node/blob/master/test/common/index.js#L580-L584) and [`common`.mustCall()`](https://github.com/nodejs/node/blob/master/test/common/index.js#L498-L533) from Node might be a better way of asserting correct asynchronous behaviour.

Reply:
By using, do you mean copying the functions to this project? Because it's not exported by Node, is it?

Reply:
Yeah, we'd pretty much have to copy them. TBH, we should be copying the whole test/addons-napi directory, because we are diverging somewhat from the implementation present in Node.js, so we need to make sure that those tests pass too. Of course that's waaay beyond the scope of this PR, but that one piece of functionality (`mustCall()` and `mustNotCall()`) would come in handy already.

Reply:
I'd suggest that copying those might make sense, I don't think we necessarily need to delay this PR for that unless @rolftimmermans believes its relatively easy to do.

Reply:
Sure, I added the test helpers that are useful for this test case. I imagine it only makes sense to add others once they're needed.